luxonis_ml/data/datasets/luxonis_dataset.py

```diff
@@ -303,105 +262,37 @@ def update_source(self, source: LuxonisSource) -> None:
         @param source: The new L{LuxonisSource} to replace the old one.
         """
 
-        self.datasets[self.dataset_name]["source"] = self._source_to_document(source)
+        self.datasets[self.dataset_name]["source"] = source.to_document()
         self._write_datasets()
         self.source = source
 
     def set_classes(self, classes: List[str], task: Optional[str] = None) -> None:
-        """Sets the names of classes for the dataset. This can be across all CV tasks or
-        certain tasks.
-
-        @type classes: List[str]
-        @param classes: List of class names to set.
-        @type task: Optional[str]
-        @param task: Optionally specify the LabelType where these classes apply.
-        """
-
         if task is not None:
-            if task not in LABEL_TYPES:
-                raise Exception(f"Task {task} is not a supported task")
             self.datasets[self.dataset_name]["classes"][task] = classes
         else:
-            for task in LABEL_TYPES:
-                self.datasets[self.dataset_name]["classes"][task] = classes
+            raise NotImplementedError(
+                "Setting classes for all tasks not yet supported. "
+                "Set classes individually for each task"
+            )
```

Comment:
Should we change `task` to be required instead?

Reply:
Same as above